annotation-processor/src/main/java/org/hibernate/validator/ap/checks/annotationparameters/AnnotationMessageCheck.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.ap.checks.annotationparameters;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.regex.Pattern;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.Element;
+
+import org.hibernate.validator.ap.checks.ConstraintCheckIssue;
+import org.hibernate.validator.ap.checks.ConstraintCheckWarning;
+import org.hibernate.validator.ap.util.AnnotationApiHelper;
+import org.hibernate.validator.ap.util.CollectionHelper;
+
+/**
+ * Checks that the message provided as annotation parameter is valid and gives a warning otherwise.
+ *
+ * @author Marko Bekhta
+ */
+public class AnnotationMessageCheck extends AnnotationParametersAbstractCheck {
+
+	// for "long ish without space"
+	private static final int LONG_MESSAGE_SIZE = 35;
```

Comment:
I'm not convinced of this one really. Whatever threshold we take, it seems arbitrary. I'd limit it to the next one.